helix-core/src/syntax.rs

```diff
@@ -1809,24 +1824,65 @@ impl<'a> Iterator for HighlightIter<'a> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum InjectionCapture<'a> {
+    Name(Cow<'a, str>),
+    File(Cow<'a, Path>),
```

Comment:
What do you think about "LanguageName" and "Filename" for these two variants? I think "File" has some connotation that it's a file handle. Or maybe "Path"?

Reply:
We could maybe also rename the enum to `InjectionLanguage`? Maybe that's more descriptive in general.

Reply:
What do you think about `InjectionLanguageMarker`?